### ISSUE TITLE
Cannot see Delegate Test to Gradle option in VS Code

### ISSUE THREAD

**Issue by jdneo** [microsoft/vscode-gradle#1606]
@jdneo I don't see Delegate Test to Gradle option in my VS Code:
![Screenshot 2024-09-13 at 17 36 30](https://github.com/user-attachments/assets/af211f65-13d0-46da-912e-cdeb87260e8e)
![Screenshot 2024-09-13 at 17 38 08](https://github.com/user-attachments/assets/91514953-f81f-4418-a703-22d1919f7da3)
![Screenshot 2024-09-13 at 17 37 01](https://github.com/user-attachments/assets/9f563d17-7c89-4a36-a495-c5797cc47ccf)

![Screenshot 2024-09-13 at 17 39 07](https://github.com/user-attachments/assets/0234fc02-4145-4750-9b38-d3b6028b8a47)
![Screenshot 2024-09-13 at 17 39 40](https://github.com/user-attachments/assets/eb61a76a-1b1d-40cc-9d38-273324622b5e)

Any hints how to enable that feature?

_Originally posted by @piotr-rachwalik in https://github.com/microsoft/vscode-java-test/issues/1045#issuecomment-2349249731_

**Comment by piotr-rachwalik** [comment]
@jdneo I checked and this feature works on my Windows machine. On my Mac, it doesn't work. Perhaps I need to reinstall the VS Code to make this work.

**Comment by piotr-rachwalik** [comment]
@jdneo I've changed this setting 
![Screenshot 2024-09-16 at 09 48 07](https://github.com/user-attachments/assets/19e93773-9b66-4e83-a4cd-37169876ece1)
and now I can see the Delegate to Gradle test option.

But when I try to run the test, I'm getting this error:
```[Error - 09:47:03] Sep 16, 2024, 9:47:03 AM Error in calling delegate command handler
Project is not a Gradle build server project: <my_project>
java.lang.IllegalArgumentException: Project is not a Gradle build server project: <my_project>
	at com.microsoft.gradle.bs.importer.handler.GradleDelegateCommandHandler.executeCommand(GradleDelegateCommandHandler.java:40)
	at org.eclipse.jdt.ls.core.internal.handlers.WorkspaceExecuteCommandHandler$1.run(WorkspaceExecuteCommandHandler.java:230)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:47)
	at org.eclipse.jdt.ls.core.internal.handlers.WorkspaceExecuteCommandHandler.executeCommand(WorkspaceExecuteCommandHandler.java:220)
	at org.eclipse.jdt.ls.core.internal.handlers.JDTLanguageServer.lambda$4(JDTLanguageServer.java:606)
	at org.eclipse.jdt.ls.core.internal.BaseJDTLanguageServer.lambda$0(BaseJDTLanguageServer.java:87)
	at java.base/java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:646)
	at java.base/java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:483)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
```

I also tried with on/off setting of build gradle server, but the error is still the same:
![Screenshot 2024-09-16 at 09 55 31](https://github.com/user-attachments/assets/69cbe813-c945-49ed-a202-ca66834839e8)

**Comment by piotr-rachwalik** [comment]
@jdneo perhaps it is because, the delegate test to gradle option does not honor the settings.gradle, where I have the settings:
```
rootProject.name = '<root_project_name>'
def subProjectsDir = new File(rootDir, 'subprojects')
fileTree(subProjectsDir) {
  include '**/build.gradle'
  exclude '**/bin/**/build.gradle'
  exclude '**/build/**/build.gradle'
  exclude '**/src/**/build.gradle'
}.collect {
  subProjectsDir.toPath().relativize(it.parentFile.toPath()).toString()
}.each {
  def projectName = ':' + rootProject.name + '-' + it.replace(File.separator, '-')
  include(projectName)
  project(projectName).projectDir = new File(subProjectsDir, it)
}
```

Please note that when I use Eclipse IDE with Buildship then I'm able to run a test using gradle.

**Comment by jdneo** [comment]
@piotr-rachwalik Thank you for sharing.

The error that `Project is not a Gradle build server project` is caused by the fact that the project is not imported by gradle build server, but buildship instead. There are several cases that will lead to this.

Have you tried running `Java: Clean Java Language Server Workspace`? If does vscode command does not help, would you mind sharing the folder structure of your project?

**Comment by piotr-rachwalik** [comment]
@jdneo I've tried running `Java: Clean Java Language Server Workspace` but it didn't help.
Here is my folder structure:
```
/build.gradle
/settings.gradle
/.gitignore
/subprojects/
   /project1/src/
                       main/java/
                       main/resources/
                       test/java/
                       test/resources/
                       build.gradle
```
and so we have multiple gradle projects in the subprojects directory.

**Comment by piotr-rachwalik** [comment]
@jdneo I think the root cause might be the `./gradlew eclipse` command that I'm using to make my project usable in Eclipse / VS Code. Without that, the dependencies are not resolved, etc.
I think that command builds the projects into Buildship readable format.

**Comment by jdneo** [comment]
I guess I may found the root cause.

The gradle extension needs to register its gradle test runner to the test runner for java extension via the api exported by test runner for java. Thus, the gradle extension will try to activate the test runner extension and then call those apis.

The problem is that, the test runner extension is depend on `redhat.java` extension. Its initialization work needs to be done after the java extension is ready. Turns out that VS Code has some issue that when activate test runner extension via the extension api, vs code won't activate the java extension (Or, vs code won't wait until java extension is ready). So the registration is actually failed.

The fix for now can be that we manually activate java extension then the test runner extension, which guarantees that all the required components are ready.

**Comment by guest73** [comment]
Hi. I am also facing this problem. When do you plan to release this fix?
Or can I build this extension (any docs how?) myself and install with
[vsce](https://code.visualstudio.com/api/working-with-extensions/publishing-extension#installation)
```
vsce package  # Generates a .vsix file
code --install-extension my-extension-0.0.1.vsix
```
?

**Comment by rjschafer66** [comment]
Hi @jdneo ,

I know this issue has been resolved, but I had the same symptom (cannot see 'Delegate to Gradle' option in the Java Test Runner).  I was able to resolve it, and thought you would be interested in what I found.
It turns out that I had both the `redhat.java` extension and the `oracle.java` extensions installed.   I did not realize this and was not experiencing conflicts until I tried to find the 'delegate to gradle' feature.
I uninstalled `oracle.java`, and now everything works as expected.  

hopefully this helps anyone else facing a similar issue.
